lib/ansible/modules/storage/purestorage/purefa_facts.py

```diff
@@ -590,7 +590,9 @@ def generate_host_dict(array):
             'iqn': hosts[host]['iqn'],
             'wwn': hosts[host]['wwn'],
             'personality': array.get_host(hostname,
-                                          personality=True)['personality']
+                                          personality=True)['personality'],
+            'target_port': array.get_host(hostname,
+                                          all=True)[0]['target_port']
```

Comment:
The 0 is assuming there is at least 1 results.  I get an Error List index out of range when I run this.

Here are the raw results returned from my array.  As you see some have empty target ports which generates this error.  And some have multiple ports.  I think the return type should just be a list and pass that directly back as is.  So, just remove the [0] from the line:

```
[
  {
    "host_wwn": null,
    "name": "esxi1",
    "host_iqn": "iqn.1998-01.com.vmware:esxi1-32e248c8",
    "vol": "FlashStack-Datastore2",
    "hgroup": "FlashStack-Cluster",
    "target_port": [],
    "lun": 254
  },
  {
    "host_wwn": "20000025B5000A02",
    "name": "esxi1",
    "host_iqn": null,
    "vol": "FlashStack-Datastore2",
    "hgroup": "FlashStack-Cluster",
    "target_port": [
      "CT1.FC2",
      "CT0.FC2"
    ],
    "lun": 254
  },
```

Reply:
Removing the 0 doesn't work. We have to pass an index. Working on a fix.